GiniVision/Classes/AnalysisContainerViewController.swift

```diff
@@ -98,9 +104,29 @@ extension AnalysisContainerViewController: AnalysisDelegate {
     
     func displayError(withMessage message: String?, andAction action: NoticeAction?) {
         let notice = NoticeView(text: message ?? "", noticeType: .error, action: action)
-        DispatchQueue.main.async { 
+        DispatchQueue.main.async {
             self.showNotice(notice)
         }
     }
     
+    func displayNoResultsScreen(completion: ((_ shown: Bool) -> ())) {
+        guard let giniNavController = self.navigationController as? GiniNavigationViewController else {
+            completion(false)
+            return
+        }
+        
+        if document.type == .image {
+            let isCameraViewControllerLoaded = giniNavController.viewControllers.contains(where: { viewController in
+                return viewController is CameraContainerViewController
+            })
+        
+            giniNavController.pushViewController(ImageAnalysisNoResultsContainerViewController(canGoBack: isCameraViewControllerLoaded), animated: true)
```

Comment:
Another one... and I really dislike that `canGoBack `.
**It's not the view controller's responsibility to know if it can go back or not.**

Reply:
It is the only way using the Screen API, using the containers...